thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datalayer/pojo/DetectionConfigBean.java

```diff
@@ -162,6 +162,12 @@ public DetectionHealth getHealth() {
   }
 
   public void setHealth(DetectionHealth health) {
+    // If there is no success execution in the current window then the last success task execution time would be -1L.
+    // We should keep the original last success task execution time if we have.
+    long lastSuccessTaskExecutionTime = this.health.getDetectionTaskStatus().getLastSuccessTaskExecutionTime();
```

Comment:
Shall we put this logic in `MonitorTaskRunner` where the detection health is updated?

Reply:
Makes sense! I changed that. Can you review again? @jihaozh